text/0000-structural-records.md

```diff
@@ -0,0 +1,1170 @@
+we did not have to define a type `MyRecordType` ahead of time.
+
+If you already had variables named `alpha` and `beta` in scope,
+just as you could have with `MyRecordType`, you could have also written:
+
+```rust
+let alpha = 42u8;
+let beta = true;
+
+let my_record = {
+    alpha,
+    beta,
+    gamma: "Dancing Ferris"
+};
+```
+
+### Pattern matching
+
+Once you have produced a structural record, you can also pattern match on the
+expression. To do so, you can write:
+
+```rust
+match my_record {
+    { alpha, beta, gamma } => println!("{}, {}, {}", alpha, beta, gamma),
+}
+```
+
+This pattern is *irrefutable* so you can also just write:
+
+```rust
+let { alpha, beta, gamma } = my_record;
+println!("{}, {}, {}", alpha, beta, gamma);
+```
+
+This is not particular to `match` and `let`. This also works for `if let`,
+`while let`, `for` loops, and function arguments.
+
+If we had used `MyRecordType`, you would have instead written:
+
+```rust
+let MyRecordType { alpha, beta, gamma } = my_record_nominal;
+println!("{}, {}, {}", alpha, beta, gamma);
+```
+
+When pattern matching on a structural record, it is also possible to give
+the binding you've created a different name. To do so, write:
+
+```rust
+let { alpha: new_alpha, beta, gamma } = my_record;
+println!("{}, {}, {}", new_alpha, beta, gamma);
+```
+
+In this snippet, we've bound the field `alpha` to the binding `new_alpha`.
+This is not limited to one field, you can do this will all of them.
+
+As with nominal structs, it's possible to ignore some or all fields when
+pattern matching on structural records. Examples include:
+
+```rust
+let { alpha, .. } = my_record;
+println!("{}", alpha);
+
+let { .. } = my_record;
+```
+
+### Field access
+
+Given the binding `my_record_nominal` of type `MyRecordType`, you can access its
+fields with the usual `my_record_nomina.alpha` syntax. This also applies to
+structural records. It is perfectly valid to move or copy:
+
+```rust
+println!("The answer to life... is: {}", my_record.alpha);
+```
+
+or to borrow a field:
+
+```rust
+fn borrow(x: &bool) { .. }
+
+borrow(&my_record.beta);
+```
+
+including mutably:
+
+```rust
+fn mutably_borrow(x: &mut bool) { .. }
+
+let mut my_record = { alpha: 42u8, beta: true, gamma: "Dancing Ferris" };
+
+mutably_borrow(&mut my_record.beta);
+```
+
+### One-field records
+
+Positional tuples with a single element use the syntax `(x,)` both for types
+and for values to distinguish between the 1-element tuple and wrapping a value
+in parenthesis. To avoid ambiguities with the `{ x }` syntax that is already
+allowed, we impose the same rule for one-field structural records. To create
+such a record, you must write `{ x, }` or `{ x: <value>, }`. This difference
+is also imposed to maintain a sense of consistency between the positional
+and the non-positional structural variant.
+
+### Struct update syntax
+
+Nominal structs support what is referred to as the *"struct update syntax",
```

Comment:
you forgot to close the star.

```suggestion
Nominal structs support what is referred to as the *"struct update syntax"*,
```

Reply:
Thanks! I'm loving this new "apply suggestion" feature!